libp2p/Host.cpp

```diff
@@ -351,15 +353,18 @@ void Host::runAcceptor()
 			}
 			
 			// asio doesn't close socket on error
-			if (!success && s->is_open())
+			if (!success)
 			{
-				boost::system::error_code ec;
-				s->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
-				s->close();
+				if (s->is_open())
+				{
+					boost::system::error_code ec;
+					s->shutdown(boost::asio::ip::tcp::socket::shutdown_both, ec);
+					s->close();
+				}
+				delete s;
```

Comment:
Promising again, that I'll move this socket into it's own safe and secure class which follows RAII.